llvm/lib/Transforms/Vectorize/SandboxVectorizer/SandboxVectorizer.cpp

```diff
@@ -52,5 +99,16 @@ bool SandboxVectorizerPass::runImpl(Function &LLVMF) {
   // Create SandboxIR for LLVMF and run BottomUpVec on it.
   sandboxir::Context Ctx(LLVMF.getContext());
   sandboxir::Function &F = *Ctx.createFunction(&LLVMF);
-  return BottomUpVecPass.runOnFunction(F);
+  if (UseRegionsFromMetadata) {
+    SmallVector<std::unique_ptr<sandboxir::Region>> Regions =
+        sandboxir::Region::createRegionsFromMD(F);
+    for (auto &R : Regions) {
+      RPM.runOnRegion(*R);
+    }
+    return false;
```

Comment:
I am not following the various discussions.  Why can't this be another function pass called "TestsFuncPass" which then runs the various regions?

Reply:
It *can*. I'm just having some trouble figuring out what we *want*.